lib/classes/ConfigSchemaHandler/index.js

```diff
@@ -290,6 +309,21 @@ class ConfigSchemaHandler {
 }
 
 function addPropertiesToSchema(subSchema, extension = { properties: {}, required: [] }) {
+  let collidingExtensionPropertyKey;
+  const existingSubSchemaPropertiesKeys = Object.keys(subSchema.properties);
+  Object.keys(extension.properties).some(extensionPropertiesKey => {
+    const isColliding = existingSubSchemaPropertiesKeys.includes(extensionPropertiesKey);
+    if (isColliding) collidingExtensionPropertyKey = extensionPropertiesKey;
+    return isColliding;
+  });
+
+  if (collidingExtensionPropertyKey) {
+    throw new this.serverless.classes.Error(
+      `Property '${collidingExtensionPropertyKey}' already have a definition - this property might have already been defined by the Serverless framework or one other plugin`,
```

Comment:
In this case _property key_ is provided without context, and it might be no clear for user definition of which exactly property throws.

I think proper way to address that would be:
1. Crash with more generic error here, and expose on `error` a property name for which we're crashing
2. In method where we use this util, catch the error and throw more meaningful one exposing full path to error